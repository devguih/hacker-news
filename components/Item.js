import React from 'react'
import Firebase from '../services/Firebase'
import { asyncReactor } from 'async-reactor';

const Loading = () => <h2>Carregando...</h2>

const Item = async ({ itemID }) => {
  // eslint-disable-next-line no-undef
  const fs = Firebase.child(`item/${itemID}`).once('value')
  const res = await fs
  const item = res.val()

  return (
    <div>
      <h2>
        <a href={item.url}>
          {item.title}
        </a>
      </h2>
      <p>{`@${item.by}`}</p>
      <small>Comments: {item.kids.length}</small>
      <hr />
    </div>
  )
}

export default asyncReactor(Item, Loading)
