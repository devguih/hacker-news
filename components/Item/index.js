import React from 'react'
import Link from 'next/link'

export default ({ item }) => (
  <article className="media">
    <figure className="media-left">
      <p className="image is-48x48 has-text-centered is-size-4">
        { item.points }
      </p>
    </figure>
    <div className="media-content is-visible">
      <div className="content">
        <h2 className="title is-4">
          <Link href={ item.url }>
            <a target="_blank">{ item.title }</a>
          </Link>
        </h2>
        <p className="subtitle is-6">
          <Link href={`/detail/${ item.id }`}>
            <a>{ item.comments_count } comments</a>
          </Link>
          {` | `}
          <strong className="has-text-grey-dark">
            <Link href={`/user/${ item.user }`}>
              <a>{ `@${item.user}` }</a>
            </Link>
          </strong>
          {` | `}
          <small>{ item.time_ago }</small>
        </p>
      </div>
    </div>
  </article>
)
