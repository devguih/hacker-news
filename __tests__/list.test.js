/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import List from '../pages/list.js'
import Item from '../components/Item';

const itemTest = {
  "id": 16461773,
  "title": "How poverty changes your mindset",
  "points": 105,
  "user": "bryanwbh",
  "time": 1519600325,
  "time_ago": "2 hours ago",
  "comments_count": 27,
  "type": "link",
  "url": "http://review.chicagobooth.edu/behavioral-science/2018/article/how-poverty-changes-your-mind-set",
  "domain": "review.chicagobooth.edu"
}

describe('<List />', () => {
  it('renders three <Item /> components', () => {
    const wrapper = shallow(<List items={[itemTest]} />);

    expect(wrapper.find(Item).length).toBe(1)
  });
});

describe('With Snapshot Testing', () => {
  it('List shows', () => {
    const component = renderer.create(<List items={[itemTest]} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
