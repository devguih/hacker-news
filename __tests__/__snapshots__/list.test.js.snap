// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing List shows 1`] = `
<div>
  <div>
    <div>
      <div
        className="navbar"
      />
      <nav
        className="navbar is-link is-fixed-top"
      >
        <div
          className="navbar-brand"
        >
          <a
            className="navbar-item"
            href="/best"
            onClick={[Function]}
          >
            Hacker news
          </a>
          <div
            className=" navbar-burger burger"
            data-target="main-navbar"
            onClick={[Function]}
          >
            <span />
            <span />
            <span />
          </div>
        </div>
        <div
          className=" navbar-menu"
          id="main-navbar"
        >
          <div
            className="navbar-end"
          >
            <a
              className="navbar-item"
              href="/best"
              onClick={[Function]}
            >
              Best
            </a>
            <a
              className="navbar-item"
              href="/news"
              onClick={[Function]}
            >
              News
            </a>
            <a
              className="navbar-item"
              href="/show"
              onClick={[Function]}
            >
              Show
            </a>
            <a
              className="navbar-item"
              href="/ask"
              onClick={[Function]}
            >
              Ask
            </a>
            <a
              className="navbar-item"
              href="/jobs"
              onClick={[Function]}
            >
              Jobs
            </a>
          </div>
        </div>
      </nav>
    </div>
    <header
      className="hero is-link"
    >
      <div
        className="hero-body"
      >
        <div
          className="container has-text-centered"
        >
          <h1
            className="title is-capitalized"
          />
        </div>
      </div>
    </header>
  </div>
  <div
    className="section"
  >
    <div
      className="container"
    >
      <article
        className="media"
      >
        <figure
          className="media-left"
        >
          <p
            className="image is-48x48 has-text-centered is-size-4"
          >
            105
          </p>
        </figure>
        <div
          className="media-content is-visible"
        >
          <div
            className="content"
          >
            <h2
              className="title is-4"
            >
              <a
                href="http://review.chicagobooth.edu/behavioral-science/2018/article/how-poverty-changes-your-mind-set"
                onClick={[Function]}
                rel="noopener"
                target="_blank"
              >
                How poverty changes your mindset
              </a>
            </h2>
            <p
              className="subtitle is-6"
            >
              <a
                href="/detail/16461773"
                onClick={[Function]}
              >
                27
                 comments
              </a>
               | 
              <strong
                className="has-text-grey-dark"
              >
                <a
                  href="/user/bryanwbh"
                  onClick={[Function]}
                >
                  @bryanwbh
                </a>
              </strong>
               | 
              <small>
                2 hours ago
              </small>
            </p>
          </div>
        </div>
      </article>
    </div>
  </div>
  <div
    className="section"
  >
    <div
      className="container"
    >
      <div
        className="columns"
      >
        <div
          className="column is-2"
        >
          <nav
            aria-label="pagination"
            className="pagination is-right"
            role="navigation"
          />
        </div>
      </div>
    </div>
  </div>
  <footer
    className="footer"
  >
    <div
      className="container"
    >
      <div
        className="content has-text-centered"
      >
        <p>
          <strong>
            Hacker News
          </strong>
          <br />
          Copyright Â© - 
          2018
        </p>
      </div>
    </div>
  </footer>
</div>
`;
